CONTRIBUTING.md

```diff
@@ -109,42 +115,346 @@ Check out [building instructions](https://github.com/prestodb/presto/tree/master
+
+Details for each point and good commit message examples can be found on https://wiki.openstack.org/wiki/GitCommitMessages#Information_in_commit_messages
+
+* **Metadata**
+    * If the commit was to solve an Github issue, refer to it at the end of a commit message in a rfc822 header line format.\
+      E.g.\
+      Resolves: #1234\
+      See also: #1235, #1236\
+
+* **Backport Commits**
+    * Must refer to original commit ID (refer to the PR link in the PR message) or link and author in the commit message
+      ```bash
+      Fix OOM caused by foo in bar
+      
+      Foo was pack ratting ByteBuffers causing OOM.
+      
+      Cherry-pick of https://github.com/prestodb/presto/pull/18424/commits/30e36dae0cdb9debd991931370b86301cd17e261 
+      Co-authored-by: Foo Bar <foo@bar.com>
+      ```
+    * Try to follow the original PR/commits structure. Do not mingle multiple original commits or reorder them.
+    * Do not add big modifications in the same backported commit, unless it’s the minimum required to resolve conflicts.
+    * If needed, add new separate commits following the backport commit for necessary changes.
+
+
 
 ## Committers
 
 Presto committers are defined as [code owners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) and documented in the project's [`CODEOWNERS`](CODEOWNERS) file.  Each line in the `CODEOWNERS` file defines a module or submodule that the committer has the rights to approve.  New modules and submodules for CODEOWNERS may be added as needed.
 
 New committers are approved by majority vote of the TSC ([see TSC charter](https://github.com/prestodb/tsc/blob/master/CHARTER.md)).  To become a committer, reach out to an [existing TSC member](https://github.com/prestodb/tsc#members) and ask for their feedback on your eligibility (see: [How to become a Presto Committer?](https://github.com/prestodb/presto/wiki/How-to-become-a-Presto-committer%3F)).  Note: to expedite the process, consider creating a document that outlines your Github stats, such as the number of reviews, lines of code added, number of PRs, and outlines particularly outstanding code and review contributions.  If the TSC member believes you are eligible, they will submit your nomination to a vote by the TSC, typically in the form of a PR that adds your handle to the `CODEOWNERS` file.  The process is complete once the PR is merged.
 
+## Pull Requests
+* #### PR size and structure
+    * A PR can consist of multiple small commits, preferably not exceeding 20(?).
+    * The total number of lines modified in a single PR shall not exceeding 5000(?)
+    * The commits sequence shall be in the dependencies order, not in date created/modified order
+    * Every commit in the PR shall pass all the tests
+      Example: https://github.com/prestodb/presto/pull/12991/commits
+
+* #### PR title
+    * Please refer to the commit title
```

Comment:
It might be worth linking to the relevant section here or maybe just add the word "above" somewhere in this line

```suggestion
    * Please refer to the section on commit subject
```